src/api/handlers/transactions.go

```diff
@@ -0,0 +1,175 @@
+package handlers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/lungria/spendshelf-backend/src/categories"
+
+	"github.com/lungria/spendshelf-backend/src/models"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/lungria/spendshelf-backend/src/transactions"
+	"go.uber.org/zap"
+)
+
+type patchCategoryRequest struct {
+	CategoryID string `json:"category_id" binding:"required"`
+}
+
+type getTransactionsResponse struct {
+	Transactions []models.Transaction `json:"transactions"`
+}
+
+// TransactionsHandler is a struct which implemented by transactions handler
+type TransactionsHandler struct {
+	txnRepo transactions.Repository
+	ctgRepo categories.Repository
+	logger  *zap.SugaredLogger
+}
+
+// NewTransactionsHandler create a new instance of TransactionsHandler
+func NewTransactionsHandler(txnRepo transactions.Repository, ctgRepo categories.Repository, logger *zap.SugaredLogger) (*TransactionsHandler, error) {
+	if txnRepo == nil || ctgRepo == nil {
+		return nil, errors.New("repository must not be nil")
+	}
+
+	if logger == nil {
+		return nil, errors.New("logger must not be nil (Transactions)")
+	}
+
+	return &TransactionsHandler{
+		txnRepo: txnRepo,
+		ctgRepo: ctgRepo,
+		logger:  logger,
+	}, nil
+}
+
+// HandleGet can return all transactions, only categorized transactions, only uncategorized transactions and transactions interrelated with one category.
+// /transactions or /transactions?category= returned all transactions
+// /transactions?category=with returned all categorized transactions
+// /transactions?category=without returned all uncategorized transactions
+// /transactions?category=categoryID returned all transactions which related with one specify category
+func (handler *TransactionsHandler) HandleGet(c *gin.Context) {
+	category := c.Request.URL.Query().Get("category")
+
+	switch category {
+	case "":
+		handler.allTransactions(c)
+		return
+	case "without":
+		handler.onlyUncategorizedTransactions(c)
+		return
+	case "with":
+		handler.onlyCategorizedTransactions(c)
+		return
+	default:
+		handler.oneCategoryTransactions(c, category)
+		return
+	}
+}
+
+// HandlePatch is setting or changing a category for specify transactionResponse
+func (handler *TransactionsHandler) HandlePatch(c *gin.Context) {
+	var req patchCategoryRequest
+	err := c.BindJSON(&req)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, errorResponse{Message: "Bad request", Error: err.Error()})
+		return
+	}
+
+	transactionID := c.Param("transactionID")
+	tObjID, err := primitive.ObjectIDFromHex(transactionID)
```

Comment:
what if transactionId == nil?

Reply:
gin returns "404 Not found" in this case